app/views/users/profile.html.erb

```diff
@@ -18,8 +18,8 @@
 		<%= f.input :phone_mobile, :label => "Phone", :input_html => { :value => number_to_phone(f.object.phone_mobile) }, :hint => "We only use your phone number to get ahold of you about events and activities that you've signed up for on this site." %>
 		<%= f.input :birthdate, :label => "Birth Date", :start_year => Time.now.year-13, :end_year => 1930 %>
         <%= f.input :other_languages, :label => "Other languages you speak", :hint => "If you speak any languages other than English, enter them here to help coordinate event planning and project assignment." %>
-	
-		<%- if @current_user.can_edit?(@person, :high_school_id) -%>
+		
+		<%- if @current_user.can_edit?(@person, :high_school_id) && @person.is_a?(Participant) -%>
```

Comment:
For some reason I was getting a nil for `high_school` in line 28 when I logged in as a Mentor regardless of the `try`. So, I added another check for a `Participant`.